bahmnicore-omod/src/main/java/org/openmrs/module/bahmnicore/web/v1_0/resource/BahmniConceptResource.java

```diff
@@ -56,35 +56,58 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 
         DelegatingResourceDescription representationDescription = super.getRepresentationDescription(rep);
         if (representationDescription == null) {
-            if (rep instanceof NamedRepresentation && rep.getRepresentation().equals("bahmni")) {
-                DelegatingResourceDescription description = new DelegatingResourceDescription();
-                description.addProperty("uuid");
-                description.addProperty("name",Representation.DEFAULT);
-                description.addProperty("names",Representation.DEFAULT);
-                description.addProperty("set");
-                description.addProperty("datatype",Representation.DEFAULT);
-                description.addProperty("conceptClass",Representation.DEFAULT);
-                description.addProperty("hiNormal");
-                description.addProperty("hiAbsolute");
-                description.addProperty("hiCritical");
-                description.addProperty("lowNormal");
-                description.addProperty("lowAbsolute");
-                description.addProperty("lowCritical");
-                description.addProperty("units");
-                description.addProperty("precise");
-                description.addProperty("allowDecimal");
-                description.addProperty("handler");
-                description.addProperty("descriptions", Representation.DEFAULT);
-                description.addProperty("answers", new NamedRepresentation("bahmniAnswer"));
-                description.addProperty("setMembers", new NamedRepresentation("bahmni"));
-                return description;
-            } else if (rep instanceof NamedRepresentation && rep.getRepresentation().equals("bahmniAnswer")) {
-                DelegatingResourceDescription description = new DelegatingResourceDescription();
-                description.addProperty("uuid", Representation.DEFAULT);
-                description.addProperty("name", Representation.DEFAULT);
-                description.addProperty("names", Representation.DEFAULT);
-                description.addProperty("displayString");
-                return description;
+            if (rep instanceof NamedRepresentation) {
```

Comment:
Means, the and part is removed - any significance to move rep.getRepresentation().equals("bahmni") inside theblock?

Reply:
Just that the condition in the and is moved inside the if - just avoiding duplication of same condition checking thrice incase where the NamedRepresentation name is bahmniFullAnswers or twice in case of bahmniAnswer